src/include/cpt/Optional.hpp

```diff
@@ -0,0 +1,41 @@
+//
+// Purpur Tentakel
+// cpp 23
+// 12.05.24
+//
+
+#pragma once
+#include <functional>
+#include <stdexcept>
+#include <variant>
+
+namespace cpt {
+    template<typename T>
+    class Otional final {
+    private:
+        std::variant<T, std::monostate> m_value;
+
+    public:
+        Otional(T value) : m_value{ std::in_place_index<0>, std::forward<T>(value) } { }
+        Otional() : m_value{ std::in_place_index<1>, std::forward<std::monostate>(std::monostate()) } { }
```

Comment:
`std::monostate` is an empty struct. It's redundant to forward it.

Reply:
so you would just do `Otional() : m_value{ std::in_place_index<1>, std::monostate() } { }` then?